device/platforms/stm32/HAL_GPIO_HANDLERS.cpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * Handles GPIO interactions for the STM HAL
+ *
+ * @author Aaron Chan
+ */
+
+#include "hashmap/hashmap.h"
+#include "device/platforms/stm32/HAL_Handlers.h"
+#include "stm32f446xx.h"
+
+namespace HALHandlers {
+    static alloc::Hashmap<GPIO_TypeDef *, dev_t, MAX_GPIO_DEVICES, MAX_GPIO_DEVICES> gpioMap;
+
+    RetType register_gpio(GPIO_TypeDef *halGPIO, CallbackDevice *dev, int num) {
+        if (gpioMap[halGPIO] != nullptr) {
+            if (!gpioMap.remove(halGPIO)) return RET_ERROR;
```

Comment:
little strange of syntax, but ok

Reply:
up to you if you want to change, but maybe make the rest of the file(s) match this syntax